plugins/importexport/doaj/classes/DOAJExportDom.inc.php

```diff
@@ -20,22 +20,35 @@ class DOAJExportDom {
 	 * Generate the export DOM tree for a given journal.
 	 * @param $doc object DOM object
 	 * @param $journal object Journal to export
+	 * @param $selectedObjects array
 	 */
-	function &generateJournalDom(&$doc, &$journal) {
-		$issueDao =& DAORegistry::getDAO('IssueDAO');
-		$sectionDao =& DAORegistry::getDAO('SectionDAO');
-		$pubArticleDao =& DAORegistry::getDAO('PublishedArticleDAO');
+	function generateJournalDom($doc, $journal, $selectedObjects) {
+		$issueDao = DAORegistry::getDAO('IssueDAO');
+		$sectionDao = DAORegistry::getDAO('SectionDAO');
+		$pubArticleDao = DAORegistry::getDAO('PublishedArticleDAO');
 
 		// Records node contains all articles, each called a record
-		$records =& XMLCustomWriter::createElement($doc, 'records');
+		$records = XMLCustomWriter::createElement($doc, 'records');
 
-		$pubArticles =& $pubArticleDao->getPublishedArticlesByJournalId($journal->getId());
-		while ($pubArticle =& $pubArticles->next()) {
-			$issue =& $issueDao->getIssueById($pubArticle->getIssueId());
+		$pubArticles = $pubArticleDao->getPublishedArticlesByJournalId($journal->getId());
+		while ($pubArticle = $pubArticles->next()) {
+
+			// check for selected issues:
+			$issueId = $pubArticle->getIssueId();
+			$selectedIssues = $selectedObjects[DOAJ_EXPORT_ISSUES];
+			if (!empty($selectedIssues) && !in_array($issueId, $selectedIssues)) continue;
+
+			$issue = $issueDao->getIssueById($issueId);
```

Comment:
I don't think there's any validation of the supplied IDs to check e.g. if they're in the current journal. This could be used to generate exports by a manager of one journal for content from another. Most DAO getter functions permit the specification of a journal ID as an optional parameter, which is often used in cases like these.